aws/resource_aws_ses_domain_identity_verification_test.go

```diff
@@ -0,0 +1,176 @@
+			{
+				Config: testAccAwsSesDomainIdentityVerification_basic(rootDomain, domain),
+				Check:  testAccCheckAwsSesDomainIdentityVerificationPassed("aws_ses_domain_identity_verification.test"),
+			},
+		},
+	})
+}
+
+func TestAccAwsSesDomainIdentityVerification_timeout(t *testing.T) {
+	domain := fmt.Sprintf(
+		"%s.terraformtesting.com",
+		acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSESDomainIdentityDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccAwsSesDomainIdentityVerification_timeout(domain),
+				ExpectError: regexp.MustCompile("Expected domain verification Success, but was in state Pending"),
+			},
+		},
+	})
+}
+
+func TestAccAwsSesDomainIdentityVerification_nonexistent(t *testing.T) {
+	domain := fmt.Sprintf(
+		"%s.terraformtesting.com",
+		acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSESDomainIdentityDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccAwsSesDomainIdentityVerification_nonexistent(domain),
+				ExpectError: regexp.MustCompile(fmt.Sprintf("SES Domain Identity %s not found in AWS", domain)),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsSesDomainIdentityVerificationPassed(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("SES Domain Identity not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("SES Domain Identity name not set")
+		}
+
+		domain := rs.Primary.ID
+		awsClient := testAccProvider.Meta().(*AWSClient)
+		conn := awsClient.sesConn
+
+		params := &ses.GetIdentityVerificationAttributesInput{
+			Identities: []*string{
+				aws.String(domain),
+			},
+		}
+
+		response, err := conn.GetIdentityVerificationAttributes(params)
+		if err != nil {
+			return err
+		}
+
+		if response.VerificationAttributes[domain] == nil {
+			return fmt.Errorf("SES Domain Identity %s not found in AWS", domain)
+		}
+
+		if *response.VerificationAttributes[domain].VerificationStatus != "Success" {
+			return fmt.Errorf("SES Domain Identity %s not successfully verified.", domain)
+		}
+
+		expected := arn.ARN{
+			AccountID: awsClient.accountid,
+			Partition: awsClient.partition,
+			Region:    awsClient.region,
+			Resource:  fmt.Sprintf("identity/%s", domain),
+			Service:   "ses",
+		}
+
+		if rs.Primary.Attributes["arn"] != expected.String() {
+			return fmt.Errorf("Incorrect ARN: expected %q, got %q", expected, rs.Primary.Attributes["arn"])
+		}
+
+		return nil
+	}
+}
+
+func testAccAwsSesDomainIdentityVerification_basic(rootDomain string, domain string) string {
+	return fmt.Sprintf(`
+resource "aws_route53_zone" "zone" {
```

Comment:
Since we need to wire this to an existing public Route53 zone that is tied to an actual registrar with the correct NS records, this should be the data source instead of the resource. 👍